onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder_helpers.h

```diff
@@ -25,15 +25,14 @@ Status AddScalarOperand(ModelBuilder& model_builder, std::vector<uint32_t>& inpu
 // adds ANEURALNETWORKS_TRANSPOSE operation
 Status AddNnapiTranspose(ModelBuilder& model_builder,
                          const std::string& data_input,
-                         const std::string& perm_input,
-                         const std::vector<int32_t>& perm,
-                         const std::string& output);
+                         const std::string& perm_input, const gsl::span<const int32_t> perm,
+                         const std::string& output, std::optional<Shape> output_shape);
 
```

Comment:
I think you can remove the output_shape arg from both AddNnapiTranspose and AddNnapiReshape.

The code in the implementation for when the input arg is nullptr matches the old behavior - namely it updates a value in Shaper with the transposed/reshaped value. I believe you could remove the arg, and always update the value in Shaper.

The calling code is either not passing in the output_arg, or it's not doing anything with it after the call. i.e. it doesn't care.